File: server/lib/launch/logger.rb
#==============================================================================
# Copyright (C) 2012-2013,2015 Stephen F. Norledge and Alces Flight Ltd.
#
# This file is part of Alces Launch.
#
# All rights reserved, see LICENSE.txt.
#==============================================================================
require 'alces/tools/logging'
require 'alces/log_writer'

module Launch
  module Logger
    extend Alces::Tools::Logging::ClassMethods

    class << self
      def log
        Alces::LogWriter.log('launch')
      end

#       def cache_warner(store, message, exc)
#         return if store.offline?
#         case exc
#         when Redis::CannotConnectError
#           Alces.app.logger.warn(message){"#{exc.class.name}: #{exc.message}"}
#         when NilClass
#           Alces.app.logger.warn(message)
#         else
#           Alces.app.logger.warn(message){exc}
#         end
#       end
    end

    if Rails.env.development?
      logger.level = Alces::Tools::Logger::DEBUG
    else
      logger.level = Alces::Tools::Logger::INFO
    end
    self.log_format(:full)
    # XXX :-p
    self.logger.instance_variable_set(:@progname,'')
  end
end

File: client/src/middleware/logger.js
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Launch.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import { createLogger } from 'redux-logger';

function configureLoggerMiddleware() {
  if (process.env.NODE_ENV === 'production') {
    // We don't want logging in production.  We aren't going to be able to
    // access the logs so they don't really do us any good.
    const identityMiddleware = () => nextMiddleware => action => nextMiddleware(action);
    return identityMiddleware;
  }

  const options = {
    collapsed: true,
    duration: true,
    level: 'info',
  };
  return createLogger(options);
}

const middleware = configureLoggerMiddleware();
export default middleware;
